tls/s2n_crypto.h

```diff
@@ -71,6 +71,7 @@ struct s2n_crypto_parameters {
     struct s2n_dh_params server_dh_params;
     struct s2n_ecc_evp_params server_ecc_evp_params;
     struct s2n_ecc_evp_params client_ecc_evp_params[S2N_ECC_EVP_SUPPORTED_CURVES_COUNT];
+    struct s2n_ecc_evp_params mutually_supported_groups[S2N_ECC_EVP_SUPPORTED_CURVES_COUNT];
```

Comment:
is there a way we could avoid adding a property? otherwise it would be good not to duplicate `s2n_ecc_evp_params` as it seems a list of s2n_ecc_named_curve would also do.

Reply:
I'm not sure if there is a way to avoid adding a property since we want to add values to the list when we parse extensions and reference the list when we select a keyshare/group? But I agree that we could just store a list of iana_ids or s2n_ecc_named_curves